Sync/StorageClient.swift

```diff
@@ -605,6 +622,80 @@ public class Sync15StorageClient {
     }
 }
 
+public class Sync15BatchClient<T: CleartextPayloadJSON> {
+    let batchToken: BatchToken
+    private let collectionURI: NSURL
+    private let postClient: POSTClient15<T>
+
+    init(batchToken: BatchToken, collectionURI: NSURL, postClient: POSTClient15<T>) {
+        self.batchToken = batchToken
+        self.collectionURI = collectionURI
+        self.postClient = postClient
+    }
+
+    func uploadRecords(lines: [String], ifUnmodifiedSince: Timestamp?) -> Deferred<Maybe<StorageResponse<POSTResult>>> {
+        let batchURI = collectionURI.URLByAppendingPathComponent("batch=\(batchToken)")
```

Comment:
Is `URLByAppendingPathComponent` correct? It's supposed to be a query parameter… and `UBAPC` encodes its argument.